src/Stratis.Bitcoin.Features.Api/Controllers/DashboardController.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.IO;
+using System.Reflection;
+using System.Text;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace Stratis.Bitcoin.Features.Api.Controllers
+{
+    /// <summary>
+    /// Controller serving static pages for the general node dashboard
+    /// </summary>
+    [Route("api/[controller]")]
+    public class DashboardController : Controller
+    {
+        private readonly ILogger logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DashboardController"/> class.
+        /// </summary>
+        public DashboardController(ILoggerFactory loggerFactory)
+        {
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+        }
+
+        /// <summary>
+        /// Serves up the requested file from the embedded resources
+        /// </summary>
+        /// <returns>Web response with text content</returns>
+        [HttpGet]
+        [Route("static/{file_name}")]
+        public ActionResult Static(string file_name)
+        {
+            var assembly = Assembly.GetEntryAssembly();
```

Comment:
We need to jump through some hoops here to read the files we want to serve. The files we want to serve are in the Features.Api project, but the hosting of files depends on the application using the Feature.Api project.  Therefore we compile the static files as "embedded resources" and need to read them from the assembly to serve them.

Reply:
Alternate approaches: 

Using MVC views instead of serving static files. The problem with this is that some dashboard builders may not want to use mvc, and instead use angular or something similar. So, perhaps make both options available - an MVC dashboard and a static pages dashboard.

Serving Static files using .UseStaticFiles() - This would work easily if the files weren't in separate assemblies from StratisD. Instead, to use this way, you have to still embed the files, but then you have to override the file provider used by the component that serves the static files. I was unable to get this working.